src/jdk.incubator.jextract/share/classes/jdk/internal/jextract/impl/RecordLayoutComputer.java

```diff
@@ -137,13 +135,8 @@ long fieldSize(Cursor c) {
         return c.isBitField() ? c.getBitFieldWidth() : c.type().size() * 8;
     }
 
-    ValueLayout bitfield(ValueLayout container, List<MemoryLayout> sublayouts) {
-        return Utils.addContents(container, MemoryLayout.ofStruct(sublayouts.toArray(new MemoryLayout[0])));
-    }
-
-    ValueLayout bitfield(long containerSize, List<MemoryLayout> sublayouts) {
-        return bitfield((ValueLayout)LayoutUtils.valueLayoutForSize(containerSize)
-                        .layout().orElseThrow(() -> new IllegalStateException("Unsupported size: " + containerSize)), sublayouts);
+    MemoryLayout bitfield(List<MemoryLayout> sublayouts) {
+        return LayoutUtils.setBitfields(MemoryLayout.ofStruct(sublayouts.toArray(new MemoryLayout[0])));
```

Comment:
Does this work correctly for unions? What if we have many bitfields that together end up being large than the union itself?

```
union Foo {
    int x;
    int bitfield1: 24;
    int bitfield2: 24;
};
```
Doesn't this get laid out like:
```
union Foo {
    int x;
    struct {
        int bitfield1: 24;
        int bitfield2: 24;
    }
};
```

With this patch? The layouts seem to differ between those two.

Reply:
My understanding is that bitfields in union are just disjoint. If you want them to be adjacent, then you need to put them inside a struct.

Reply:
I guess that's what I'm saying. The patch seems to be collecting all the bitfields at once and then emitting a struct layout that holds them right? But, since in a union bitfields overlap, it seems that by putting them adjacent in a struct like this has the potential to create a layout that is larger then the actual expected size.

Reply:
Ah, never mind. I see you are only collecting all the bitfields in the Struct case, but in the union case are just emitting a Struct layout with a single bitfield inside, more like:

```
union Foo {
    int x;
    struct {
        int bitfield1: 24;
    }
    struct {
        int bitfield2: 24;
    }
};
```